chia/consensus/blockchain.py

```diff
@@ -495,9 +498,16 @@ async def validate_unfinished_block(
 
         npc_result = None
         if block.transactions_generator is not None:
-            block_generator: Optional[BlockGenerator] = await self.get_block_generator(block)
-            assert block_generator is not None
-            npc_result = get_name_puzzle_conditions(block_generator, self.constants.MAX_BLOCK_COST_CLVM, False)
+            assert block.transactions_info is not None
+            try:
+                block_generator: Optional[BlockGenerator] = await self.get_block_generator(block)
+            except ValueError:
+                return PreValidationResult(uint16(Err.GENERATOR_REF_HAS_NO_GENERATOR.value), None, None)
+            if block_generator is None:
```

Comment:
presumably this wouldn't happen, right?

Reply:
No because the peer can send us a block with invalid generator refs, and we should error out on it